File: src/main/java/com/indiepost/model/PostTag.java
package com.indiepost.model;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.io.Serializable;

@Entity
@Table(name = "Posts_Tags")
public class PostTag implements Serializable {

    private static final long serialVersionUID = 5610937158697175938L;

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id")
    private Post post;


    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "tag_id")
    private Tag tag;

    @NotNull
    @Column(nullable = false)
    private int priority = 0;

    private PostTag() {
    }

    public PostTag(Post post, Tag tag, @NotNull int priority) {
        this.post = post;
        this.tag = tag;
        this.priority = priority;
    }

    public int getPriority() {
        return priority;
    }

    public void setPriority(int priority) {
        this.priority = priority;
    }

    public Post getPost() {
        return post;
    }

    public void setPost(Post post) {
        this.post = post;
    }

    public Tag getTag() {
        return tag;
    }

    public void setTag(Tag tag) {
        this.tag = tag;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }
}

File: src/main/kotlin/com/indiepost/model/PostTag.kt
package com.indiepost.model

import javax.persistence.*
import javax.validation.constraints.NotNull

@Entity
@Table(name = "Posts_Tags")
data class PostTag(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        var id: Long? = null,

        @NotNull
        @Column(nullable = false)
        var priority: Int = 0,

        @Column(name = "postId", insertable = false, updatable = false)
        var postId: Long? = null,

        @Column(name = "tagId", insertable = false, updatable = false)
        var tagId: Long? = null,

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "postId")
        var post: Post? = null,

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "tagId")
        var tag: Tag? = null
)

